File: sdks/python/src/echo_agent_sdk/dependency_values.py
from enum import Enum


class DependencyKind(str, Enum):
    BINARY = "binary"
    PYTHON_PKG = "python_pkg"
    NODE_MODULE = "node_module"

    def as_str(self) -> str:
        return self.value


class SkillSource(str, Enum):
    LOCAL = "local"
    MCP = "mcp"

    def as_str(self) -> str:
        return self.value

File: sdks/typescript/src/dependency_values.ts
/** Skill dependency/source values projected without probing or loading. */
export const DependencyKind = Object.freeze({
  Binary: "binary",
  PythonPkg: "python_pkg",
  NodeModule: "node_module",
} as const);

export type DependencyKind = (typeof DependencyKind)[keyof typeof DependencyKind];

export const SkillSource = Object.freeze({
  Local: "local",
  Mcp: "mcp",
} as const);

export type SkillSource = (typeof SkillSource)[keyof typeof SkillSource];
